app-backend/db/src/main/scala/com/azavea/rf/database/UserGroupRoleDao.scala

```diff
@@ -106,29 +106,69 @@ object UserGroupRoleDao extends Dao[UserGroupRole] {
       .list
   }
 
-  def listUsersByGroup(groupType: GroupType, groupId: UUID, page: PageRequest, searchParams: SearchQueryParameters): ConnectionIO[PaginatedResponse[User.WithGroupRole]] = {
+  def listUsersByGroup(groupType: GroupType, groupId: UUID, page: PageRequest, searchParams: SearchQueryParameters, actingUser: User): ConnectionIO[PaginatedResponse[User.WithGroupRole]] = {
+    // PUBLIC users can be seen by anyone within the same platform
+    // PRIVATE users can be seen by anyone within the same organization
+    // PRIVATE users can be seen by anyone within the same team (even if orgs are different)
     val sf =
-      fr"""SELECT u.id, u.role, u.created_at, u.modified_at,
-        u.dropbox_credential, u.planet_credential, u.email_notifications,
-        u.email, u.name, u.profile_image_uri, u.is_superuser, u.is_active, ugr.group_role
+      fr"""
+        SELECT u.id, u.role, u.created_at, u.modified_at,
+          u.dropbox_credential, u.planet_credential, u.email_notifications,
+          u.email, u.name, u.profile_image_uri, u.is_superuser, u.is_active, u.visibility,
+          ugr.group_role
         FROM """ ++ tableF ++ fr""" ugr
         JOIN """ ++ UserDao.tableF ++ fr""" u
           ON u.id = ugr.user_id
       """
 
     val cf =
-      fr"""SELECT count(ugr.id)
+      fr"""
+        SELECT count(ugr.id)
         FROM """ ++ tableF ++ fr"""AS ugr
         JOIN """ ++ UserDao.tableF ++ fr""" u
           ON u.id = ugr.user_id
       """
 
+    // Filter that applies the following rules:
+    // 1) A user is always allowed to list themselves
+    // 2) A public user is always visible
+    // 3) A private user is only visible to those within the same organization or team
```

Comment:
This should probably only apply if they are not in the public organization

Reply:
Consensus from standup this morning -- you can't restrict visibility if you're in the public org.